control_launch/launch/control.launch.py

```diff
@@ -53,24 +65,41 @@ def launch_setup(context, *args, **kwargs):
         lane_departure_checker_param = yaml.safe_load(f)["/**"]["ros__parameters"]
 
     # lateral controller
-    lat_controller_component = ComposableNode(
-        package="trajectory_follower_nodes",
-        plugin="autoware::motion::control::trajectory_follower_nodes::LateralController",
-        name="lateral_controller_node_exe",
-        namespace="trajectory_follower",
-        remappings=[
-            ("~/input/reference_trajectory", "/planning/scenario_planning/trajectory"),
-            ("~/input/current_odometry", "/localization/kinematic_state"),
-            ("~/input/current_steering", "/vehicle/status/steering_status"),
-            ("~/output/control_cmd", "lateral/control_cmd"),
-            ("~/output/predicted_trajectory", "lateral/predicted_trajectory"),
-            ("~/output/diagnostic", "lateral/diagnostic"),
-        ],
-        parameters=[
-            lat_controller_param,
-        ],
-        extra_arguments=[{"use_intra_process_comms": LaunchConfiguration("use_intra_process")}],
-    )
+    if lateral_controller_mode == "mpc_follower":
+        lat_controller_component = ComposableNode(
+            package="trajectory_follower_nodes",
+            plugin="autoware::motion::control::trajectory_follower_nodes::LateralController",
+            name="lateral_controller_node_exe",
+            namespace="trajectory_follower",
+            remappings=[
+                ("~/input/reference_trajectory", "/planning/scenario_planning/trajectory"),
+                ("~/input/current_odometry", "/localization/kinematic_state"),
+                ("~/input/current_steering", "/vehicle/status/steering_status"),
+                ("~/output/control_cmd", "lateral/control_cmd"),
+                ("~/output/predicted_trajectory", "lateral/predicted_trajectory"),
+                ("~/output/diagnostic", "lateral/diagnostic"),
+            ],
+            parameters=[
+                lat_controller_param,
+            ],
+            extra_arguments=[{"use_intra_process_comms": LaunchConfiguration("use_intra_process")}],
+        )
+    elif lateral_controller_mode == "pure_pursuit":
+        lat_controller_component = ComposableNode(
+            package="pure_pursuit",
+            plugin="pure_pursuit::PurePursuitNode",
+            name="pure_pursuit_node_exe",
+            namespace="trajectory_follower",
+            remappings=[
+                ("input/reference_trajectory", "/planning/scenario_planning/trajectory"),
+                ("input/current_odometry", "/localization/kinematic_state"),
+                ("output/control_raw", "lateral/control_cmd"),
+            ],
+            parameters=[
+                lat_controller_param,
+            ],
+            extra_arguments=[{"use_intra_process_comms": LaunchConfiguration("use_intra_process")}],
+        )
```

Comment:
I think using IfCondition is better.
https://github.com/tier4/AutowareArchitectureProposal_launcher/blob/5527a3b1541466aad4586128cb4d1fa715143a12/planning_launch/launch/scenario_planning/lane_driving/motion_planning/motion_planning.launch.py#L182-L185

Reply:
Fixed.
https://github.com/tier4/AutowareArchitectureProposal_launcher/pull/166/commits/9eb3b64329f38fa4386cedc4df9e19a80a031a5c